Patch-Source: https://src.fedoraproject.org/rpms/perl-DateTime-Format-Flexible/raw/rawhide/f/DateTime-Format-Flexible-0.34-Use-America-Panama-instead-of-EST-in-tests.patch

Upstream-Issue: https://rt.cpan.org/Public/Bug/Display.html?id=155591
--
From b332d4018415deedff6946cb8db602c45277c09a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 17 Sep 2024 18:08:17 +0200
Subject: [PATCH] Use America/Panama instead of EST in tests
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

tzdata-2024b added "EST" alias for "America/Panama" time zone into
"backward" file. This change was inherited to DateTime-TimeZone-2.63:

    $ perl -MDateTime::TimeZone -e 'print DateTime::TimeZone->new(name=>q{EST})->name, qq{\n}'
    America/Panama

As a result t/parsing.t started to fail on "EST" lines after upgrading
to DateTime-TimeZone-2.63:

    #   Failed test 'timezone => EST'
    #   at t/lib/helper.pm line 88.
    #          got: 'America/Panama'
    #     expected: 'EST'
    # Looks like you failed 29 tests of 8060.

This patch fixes the failures by using "America/Panama" to avoid using
identifiers which have changed to an alias.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 t/data/timezones.txt | 58 ++++++++++++++++++++++----------------------
 1 file changed, 29 insertions(+), 29 deletions(-)

diff --git a/t/data/timezones.txt b/t/data/timezones.txt
index dfd8052..449b2ee 100644
--- a/t/data/timezones.txt
+++ b/t/data/timezones.txt
@@ -251,7 +251,7 @@
 10-31-2009 => 2009-10-31T00:00:00 => floating
 10/4/2009 => 2009-10-04T00:00:00 => floating
 10/5/2009 => 2009-10-05T00:00:00 => floating
-10 Apr 2009 04:24:45 EST => 2009-04-10T04:24:45 => EST
+10 Apr 2009 04:24:45 America/Panama => 2009-04-10T04:24:45 => America/Panama
 10-Apr-2011. => 2011-04-10T00:00:00 => floating
 10 APR 2016 => 2016-04-10T00:00:00 => floating
 10-Aug-2009. => 2009-08-10T00:00:00 => floating
@@ -332,8 +332,8 @@
 12-Jan-2010. => 2010-01-12T00:00:00 => floating
 12-Jan-2012. => 2012-01-12T00:00:00 => floating
 12-Jun-2015 => 2015-06-12T00:00:00 => floating
-12 Mar 2010 04:48:37 EST => 2010-03-12T04:48:37 => EST
-12 Mar 2010 10:06:39 EST => 2010-03-12T10:06:39 => EST
+12 Mar 2010 04:48:37 America/Panama => 2010-03-12T04:48:37 => America/Panama
+12 Mar 2010 10:06:39 America/Panama => 2010-03-12T10:06:39 => America/Panama
 12 Mar 2010 12:24:39:000   UTC => 2010-03-12T12:24:39 => UTC
 12-Mar-2010 => 2010-03-12T00:00:00 => floating
 12-Mar-2010. => 2010-03-12T00:00:00 => floating
@@ -361,7 +361,7 @@
 13-MAR-2010 => 2010-03-13T00:00:00 => floating
 13 May 2009 14:34:00 => 2009-05-13T14:34:00 => floating
 13-May-2012. => 2012-05-13T00:00:00 => floating
-13 Nov 2009 04:56:16 EST => 2009-11-13T04:56:16 => EST
+13 Nov 2009 04:56:16 America/Panama => 2009-11-13T04:56:16 => America/Panama
 13-Nov-2009 => 2009-11-13T00:00:00 => floating
 13-Oct-2009. => 2009-10-13T00:00:00 => floating
 13-Sep-2009. => 2009-09-13T00:00:00 => floating
@@ -379,7 +379,7 @@
 14 Jun 2010 08:46:07 UTC => 2010-06-14T08:46:07 => UTC
 14-Jun-2010. => 2010-06-14T00:00:00 => floating
 14-Jun-2018. => 2018-06-14T00:00:00 => floating
-14 Mar 2011 00:00:00 EST => 2011-03-14T00:00:00 => EST
+14 Mar 2011 00:00:00 America/Panama => 2011-03-14T00:00:00 => America/Panama
 14-Nov-2009. => 2009-11-14T00:00:00 => floating
 14 Nov 2010 18:54:14 UTC => 2010-11-14T18:54:14 => UTC
 14-Nov-2011. => 2011-11-14T00:00:00 => floating
@@ -387,11 +387,11 @@
 14-Oct-2018. => 2018-10-14T00:00:00 => floating
 14-Sep-2009. => 2009-09-14T00:00:00 => floating
 14-Sep-2011. => 2011-09-14T00:00:00 => floating
-15 Aug 2009 08:04:50 EST => 2009-08-15T08:04:50 => EST
+15 Aug 2009 08:04:50 America/Panama => 2009-08-15T08:04:50 => America/Panama
 15-Aug-2009 => 2009-08-15T00:00:00 => floating
 15-Dec-2010. => 2010-12-15T00:00:00 => floating
 15-Feb-2010 => 2010-02-15T00:00:00 => floating
-15 Feb 2011 03:59:38 EST => 2011-02-15T03:59:38 => EST
+15 Feb 2011 03:59:38 America/Panama => 2011-02-15T03:59:38 => America/Panama
 15-Feb-2013. => 2013-02-15T00:00:00 => floating
 15-Jan-2010 => 2010-01-15T00:00:00 => floating
 15-Jul-2011. => 2011-07-15T00:00:00 => floating
@@ -421,7 +421,7 @@
 16-Jun-2009. => 2009-06-16T00:00:00 => floating
 16-Jun-2011. => 2011-06-16T00:00:00 => floating
 16-Mar-2009. => 2009-03-16T00:00:00 => floating
-16 Mar 2010 19:51:23 EST => 2010-03-16T19:51:23 => EST
+16 Mar 2010 19:51:23 America/Panama => 2010-03-16T19:51:23 => America/Panama
 16-Mar-2010 => 2010-03-16T00:00:00 => floating
 16-Mar-2010. => 2010-03-16T00:00:00 => floating
 16 May 2010 => 2010-05-16T00:00:00 => floating
@@ -448,7 +448,7 @@
 17-Jun-2010. => 2010-06-17T00:00:00 => floating
 17-Jun-2013. => 2013-06-17T00:00:00 => floating
 17-mar-2004 => 2004-03-17T00:00:00 => floating
-17 Mar 2010 15:00:00 EST => 2010-03-17T15:00:00 => EST
+17 Mar 2010 15:00:00 America/Panama => 2010-03-17T15:00:00 => America/Panama
 17-Mar-2010 => 2010-03-17T00:00:00 => floating
 17-Mar-2010. => 2010-03-17T00:00:00 => floating
 17 MAR 2012 => 2012-03-17T00:00:00 => floating
@@ -476,7 +476,7 @@
 18-Jun-2009. => 2009-06-18T00:00:00 => floating
 18-Jun-2011. => 2011-06-18T00:00:00 => floating
 18-Mar-2010 => 2010-03-18T00:00:00 => floating
-18 May 2009 14:57:55 EST => 2009-05-18T14:57:55 => EST
+18 May 2009 14:57:55 America/Panama => 2009-05-18T14:57:55 => America/Panama
 18-May-2009. => 2009-05-18T00:00:00 => floating
 18-May-2013. => 2013-05-18T00:00:00 => floating
 18-Nov-2009. => 2009-11-18T00:00:00 => floating
@@ -495,7 +495,7 @@
 19-Aug-2010 => 2010-08-19T00:00:00 => floating
 19 Dec 2009 => 2009-12-19T00:00:00 => floating
 19-feb-2004 => 2004-02-19T00:00:00 => floating
-19 Feb 2009 12:07:40 EST => 2009-02-19T12:07:40 => EST
+19 Feb 2009 12:07:40 America/Panama => 2009-02-19T12:07:40 => America/Panama
 19-Feb-2010. => 2010-02-19T00:00:00 => floating
 19-Jan-2010 => 2010-01-19T00:00:00 => floating
 19-Jul-2009 => 2009-07-19T00:00:00 => floating
@@ -515,8 +515,8 @@
 19 SEP 2009 => 2009-09-19T00:00:00 => floating
 19-Sep-2010. => 2010-09-19T00:00:00 => floating
 19-Sep-2012. => 2012-09-19T00:00:00 => floating
-1 Apr 2011 15:21:54 EST => 2011-04-01T15:21:54 => EST
-1 Jul 2011 12:30:31 EST => 2011-07-01T12:30:31 => EST
+1 Apr 2011 15:21:54 America/Panama => 2011-04-01T15:21:54 => America/Panama
+1 Jul 2011 12:30:31 America/Panama => 2011-07-01T12:30:31 => America/Panama
 2004-10-28 => 2004-10-28T00:00:00 => floating
 2007-03-17 => 2007-03-17T00:00:00 => floating
 2008-05-12 00:00:00 => 2008-05-12T00:00:00 => floating
@@ -1924,7 +1924,7 @@
 21-Apr-2009. => 2009-04-21T00:00:00 => floating
 21-Aug-2010 => 2010-08-21T00:00:00 => floating
 21-Aug-2011. => 2011-08-21T00:00:00 => floating
-21 Feb 2011 21:05:13 EST => 2011-02-21T21:05:13 => EST
+21 Feb 2011 21:05:13 America/Panama => 2011-02-21T21:05:13 => America/Panama
 21-Feb-2014. => 2014-02-21T00:00:00 => floating
 21-Jan-2010. => 2010-01-21T00:00:00 => floating
 21-Jun-2009 => 2009-06-21T00:00:00 => floating
@@ -1983,7 +1983,7 @@
 23-Oct-2011. => 2011-10-23T00:00:00 => floating
 23-Oct-2012. => 2012-10-23T00:00:00 => floating
 23-Sep-2009. => 2009-09-23T00:00:00 => floating
-23 Sep 2010 14:00:00 EST => 2010-09-23T14:00:00 => EST
+23 Sep 2010 14:00:00 America/Panama => 2010-09-23T14:00:00 => America/Panama
 24-Apr-2010. => 2010-04-24T00:00:00 => floating
 24-Aug-2009. => 2009-08-24T00:00:00 => floating
 24-Aug-2010. => 2010-08-24T00:00:00 => floating
@@ -1998,7 +1998,7 @@
 24-jan-2013 => 2013-01-24T00:00:00 => floating
 24-Jul-2009. => 2009-07-24T00:00:00 => floating
 24-Jul-2010. => 2010-07-24T00:00:00 => floating
-24 Jul 2014 10:58:44 EST => 2014-07-24T10:58:44 => EST
+24 Jul 2014 10:58:44 America/Panama => 2014-07-24T10:58:44 => America/Panama
 24-Jul-2018. => 2018-07-24T00:00:00 => floating
 24-Jun-2009 => 2009-06-24T00:00:00 => floating
 24-Mar-2010. => 2010-03-24T00:00:00 => floating
@@ -2069,7 +2069,7 @@
 27-Feb-2017. => 2017-02-27T00:00:00 => floating
 27-Jan-2010. => 2010-01-27T00:00:00 => floating
 27-Jun-2009. => 2009-06-27T00:00:00 => floating
-27 May 2010 16:54:26 EST => 2010-05-27T16:54:26 => EST
+27 May 2010 16:54:26 America/Panama => 2010-05-27T16:54:26 => America/Panama
 27-May-2012. => 2012-05-27T00:00:00 => floating
 27-May-2013. => 2013-05-27T00:00:00 => floating
 27-Nov-2009. => 2009-11-27T00:00:00 => floating
@@ -2082,7 +2082,7 @@
 28-Apr-2011. => 2011-04-28T00:00:00 => floating
 28-Apr-2012. => 2012-04-28T00:00:00 => floating
 28-APR-2012 => 2012-04-28T00:00:00 => floating
-28 Aug 2009 11:38:42 EST => 2009-08-28T11:38:42 => EST
+28 Aug 2009 11:38:42 America/Panama => 2009-08-28T11:38:42 => America/Panama
 28-Aug-2017. => 2017-08-28T00:00:00 => floating
 28-Dec-2010. => 2010-12-28T00:00:00 => floating
 28-Feb-2010 => 2010-02-28T00:00:00 => floating
@@ -2126,7 +2126,7 @@
 29-Sep-2014. => 2014-09-29T00:00:00 => floating
 29-Sep-2015. => 2015-09-29T00:00:00 => floating
 2-Sep-2009 => 2009-09-02T00:00:00 => floating
-30 Apr 2010 11:08:55 EST => 2010-04-30T11:08:55 => EST
+30 Apr 2010 11:08:55 America/Panama => 2010-04-30T11:08:55 => America/Panama
 30 AUG 2010 => 2010-08-30T00:00:00 => floating
 30-AUG-2014 => 2014-08-30T00:00:00 => floating
 30-aug-2015 => 2015-08-30T00:00:00 => floating
@@ -2146,7 +2146,7 @@
 30-Sep-2009 => 2009-09-30T00:00:00 => floating
 3/1/2010 => 2010-03-01T00:00:00 => floating
 3/15/2010 => 2010-03-15T00:00:00 => floating
-31 Aug 2009 07:22:27 EST => 2009-08-31T07:22:27 => EST
+31 Aug 2009 07:22:27 America/Panama => 2009-08-31T07:22:27 => America/Panama
 31-Aug-2009. => 2009-08-31T00:00:00 => floating
 31-Jan-2010. => 2010-01-31T00:00:00 => floating
 31-Jul-2009 04:00:00 UTC => 2009-07-31T04:00:00 => UTC
@@ -2168,11 +2168,11 @@
 3/28/2010 => 2010-03-28T00:00:00 => floating
 3/4/2010 => 2010-03-04T00:00:00 => floating
 3/5/2010 => 2010-03-05T00:00:00 => floating
-3 May 2010 20:31:27 EST => 2010-05-03T20:31:27 => EST
+3 May 2010 20:31:27 America/Panama => 2010-05-03T20:31:27 => America/Panama
 3-Nov-2009 => 2009-11-03T00:00:00 => floating
-3 Nov 2010 19:11:36 EST => 2010-11-03T19:11:36 => EST
+3 Nov 2010 19:11:36 America/Panama => 2010-11-03T19:11:36 => America/Panama
 3-Nov-2011 => 2011-11-03T00:00:00 => floating
-3 Oct 2012 16:14:14 EST => 2012-10-03T16:14:14 => EST
+3 Oct 2012 16:14:14 America/Panama => 2012-10-03T16:14:14 => America/Panama
 4/25/2009 => 2009-04-25T00:00:00 => floating
 4/8/2010 => 2010-04-08T00:00:00 => floating
 5/10/2017 => 2017-05-10T00:00:00 => floating
@@ -2181,19 +2181,19 @@
 6/20/2009 => 2009-06-20T00:00:00 => floating
 6/28/2009 => 2009-06-28T00:00:00 => floating
 6-Aug-2009 => 2009-08-06T00:00:00 => floating
-6 Sep 2009 23:07:30 EST => 2009-09-06T23:07:30 => EST
+6 Sep 2009 23:07:30 America/Panama => 2009-09-06T23:07:30 => America/Panama
 7/19/2011 3:46:36 PM => 2011-07-19T15:46:36 => floating
 7/20/2009 => 2009-07-20T00:00:00 => floating
 7/6/2011 3:18:10 PM => 2011-07-06T15:18:10 => floating
 7 Feb 2011 19:30:34 Africa/Abidjan => 2011-02-07T19:30:34 => Africa/Abidjan
-7 Feb 2011 19:30:34 EST => 2011-02-07T19:30:34 => EST
+7 Feb 2011 19:30:34 America/Panama => 2011-02-07T19:30:34 => America/Panama
 7-Feb-2011 => 2011-02-07T00:00:00 => floating
-7 Sep 2015 12:06:36 EST => 2015-09-07T12:06:36 => EST
+7 Sep 2015 12:06:36 America/Panama => 2015-09-07T12:06:36 => America/Panama
 8/16/2009 => 2009-08-16T00:00:00 => floating
-8 Aug 2009 09:51:05 EST => 2009-08-08T09:51:05 => EST
+8 Aug 2009 09:51:05 America/Panama => 2009-08-08T09:51:05 => America/Panama
 9/10/2009 => 2009-09-10T00:00:00 => floating
 9/7/2009 => 2009-09-07T00:00:00 => floating
-9 Mar 2011 21:08:06 EST => 2011-03-09T21:08:06 => EST
+9 Mar 2011 21:08:06 America/Panama => 2011-03-09T21:08:06 => America/Panama
 Apr-07-2010 => 2010-04-07T00:00:00 => floating
 Apr 09, 2011 => 2011-04-09T00:00:00 => floating
 Apr-10-2010 => 2010-04-10T00:00:00 => floating
@@ -2524,7 +2524,7 @@ Wed, Sep 26, 2012 => 2012-09-26T00:00:00 => floating
 2004-03-29T11:41:00Z => 2004-03-29T11:41:00 => UTC
 Mon, 19 Jan 1998 08:11:34 +1030 => 1998-01-19T08:11:34 => +1030
 1997020201:00:00+0100 => 1997-02-02T01:00:00 => +0100
-19980102030405 EST => 1998-01-02T03:04:05 => EST
+19980102030405 America/Panama => 1998-01-02T03:04:05 => America/Panama
 20021216090000 -0500 => 2002-12-16T09:00:00 => -0500
 5:30 pm GMT 121065 => 2065-12-10T17:30:00 => UTC
 5:30 pm GMT 12/10/65 => 2065-12-10T17:30:00 => UTC
-- 
2.46.1

